coala_quickstart/green_mode/find_globs.py

```diff
@@ -0,0 +1,185 @@
+            dir_name = dir_name.replace(os.sep + os.sep, os.sep)
+            list_of_files = list()
+            for (dirpath, dirnames, filenames) in os.walk(dir_name):
+                list_of_files += [os.path.join(dirpath, file)
+                                  for file in filenames]
+            files_in_section = list(child.get_files(dir_name))
+            number_of_common_files = 0
+            files_not_common = []
+            for file in list_of_files:
+                if file in files_in_section:
+                    number_of_common_files += 1
+                else:
+                    files_not_common.append(file)
+            # The number 7 is chosen as a plausible guess for when
+            # the number of files in the ignore
+            # section exceed the limit.
+            ratio = float(number_of_common_files/len(list_of_files))
+            if list_of_files != [] and (ratio >= 0.9 or len(
+                    files_not_common) <= 7):
+                glob = dir_name + os.sep + '**'
+                new_files = []
+                for file in files:
+                    if not fnmatch(file, glob):
+                        new_files.append(file)
+                new_files.append(glob)
+                files = new_files
+                ignore_list += files_not_common
+                # Since the glob '**' is found to be valid irresepective
+                # of the extension,
+                # all the files/leaves in the corresponding subtree are covered.
+                return files, ignore_list
+            else:
+                files_in_section_ext_dict = {}
+                for file in files_in_section:
+                    ext = os.path.splitext(file)[1]
+                    if ext in files_in_section_ext_dict:
+                        files_in_section_ext_dict[ext].append(file)
+                    else:
+                        files_in_section_ext_dict[ext] = [file]
+
+                list_of_files_ext_dict = {}
+                for file in list_of_files:
+                    ext = os.path.splitext(file)[1]
+                    if ext in list_of_files_ext_dict:
+                        list_of_files_ext_dict[ext].append(file)
+                    else:
+                        list_of_files_ext_dict[ext] = [file]
+
+                for ext in files_in_section_ext_dict:
+                    if ext in already_found_ext:
+                        continue
+                    val_in_sec = files_in_section_ext_dict[ext]
+                    val_in_files = list_of_files_ext_dict[ext]
+                    not_common_ext_files = []
+                    num_ext_common_files = 0
+                    for file in val_in_files:
+                        if file in val_in_sec:
+                            num_ext_common_files += 1
+                        else:
+                            not_common_ext_files.append(file)
+
+                    if val_in_files != [] and (
```

Comment:
This check is also redundant (like https://github.com/coala/coala-quickstart/pull/292/commits/e57c99ee9cbfea29d6dc6b15e7aa0a4012604072#diff-1d306ddb0db2c74fd31684901fe68ea3R56), move it to a small function